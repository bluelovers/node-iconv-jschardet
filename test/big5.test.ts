/**
 * Created by user on 2018/1/27/027.
 */

import * as iconvLite from 'iconv-lite';
import localDev, { relative, expect, path, assert, util } from './_local-dev';

import * as self from '../';

// @ts-ignore
describe(relative(__filename), () =>
{
	let currentTest;

	/**
	 * 次常用國字標準字體表
	 */
	const BIG5_STR = "\xa6\xb8\xb1\x60\xa5\xce\xb0\xea\xa6\x72\xbc\xd0\xb7\xc7\xa6\x72\xc5\xe9\xaa\xed";
	const BIG5_BUF = Buffer.from('a6b8b160a5ceb0eaa672bcd0b7c7a672c5e9aaed', 'hex');

	const BIG5_UTF8_BUF = Buffer.from('e6 ac a1 e5 b8 b8 e7 94 a8 e5 9c 8b e5 ad 97 e6 a8 99 e6 ba 96 e5 ad 97 e9 ab 94 e8 a1 a8'.replace(/\s/g, ''), 'hex');

	self.skipDecodeWarning();

	beforeEach(function ()
	{
		currentTest = this.currentTest;

		//console.log('it:before', currentTest.title);
		//console.log('it:before', currentTest.fullTitle());
	});

	describe(`init`, () =>
	{
		it(`BIG5_STR`, function (done)
		{
			let c = self.detect(BIG5_STR);

			expect(c.name).to.be.equal('Big5');
			expect(c.encoding).to.be.equal('Big5');

			done();
		});

		it(`BIG5_BUF`, function (done)
		{
			let c = self.detect(BIG5_BUF);

			expect(c.name).to.be.equal('Big5');
			expect(c.encoding).to.be.equal('Big5');

			done();
		});
	});

	describe(`iconv`, () =>
	{
		it(`encode big5`, function (done)
		{
			//console.log('it:inner', currentTest.title);
			//console.log('it:inner', currentTest.fullTitle());

			//expect(r).to.be.ok;
			//assert.isOk(r.value, util.inspect(r));

			let data = self.encode(BIG5_STR, 'big5');

			let c = self.detect(data);

			expect(c.name).to.be.equal('Big5');
			expect(c.encoding).to.be.equal('Big5');

			done();
		});
		it(`encode big5`, function (done)
		{
			//console.log('it:inner', currentTest.title);
			//console.log('it:inner', currentTest.fullTitle());

			//expect(r).to.be.ok;
			//assert.isOk(r.value, util.inspect(r));

			let data = self.encode(BIG5_BUF, 'big5');

			let c = self.detect(data, true);

			expect(c.name).to.be.equal('Big5');
			expect(c.encoding).to.be.equal('Big5');

			done();
		});

		it(`encode`, function (done)
		{
			let data = self.encode(BIG5_STR);

			let c = self.detect(data);

			expect(c.name).to.be.equal('UTF-8');
			expect(c.encoding).to.be.equal('UTF-8');

			expect(data).to.be.deep.equal(BIG5_UTF8_BUF);

			done();
		});

		it(`encode`, function (done)
		{
			//console.log('it:inner', currentTest.title);
			//console.log('it:inner', currentTest.fullTitle());

			//expect(r).to.be.ok;
			//assert.isOk(r.value, util.inspect(r));

			let data = self.encode(BIG5_BUF);

			let c = self.detect(data);

			expect(c.name).to.be.equal('UTF-8');
			expect(c.encoding).to.be.equal('UTF-8');

			expect(data).to.be.deep.equal(BIG5_UTF8_BUF);

			done();
		});
	});
});
