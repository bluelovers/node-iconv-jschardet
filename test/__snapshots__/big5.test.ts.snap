// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check encoding BIG5_BUF 1`] = `
Object {
  "confidence": 0.99,
  "encoding": "Big5",
  "name": "Big5",
}
`;

exports[`check encoding BIG5_STR 1`] = `
Object {
  "confidence": 0.99,
  "encoding": "Big5",
  "name": "Big5",
}
`;

exports[`iconv encode 1`] = `
Object {
  "confidence": 0.99,
  "encoding": "UTF-8",
  "name": "UTF-8",
}
`;

exports[`iconv encode 2`] = `
Object {
  "data": Array [
    230,
    172,
    161,
    229,
    184,
    184,
    231,
    148,
    168,
    229,
    156,
    139,
    229,
    173,
    151,
    230,
    168,
    153,
    230,
    186,
    150,
    229,
    173,
    151,
    233,
    171,
    148,
    232,
    161,
    168,
  ],
  "type": "Buffer",
}
`;

exports[`iconv encode 3`] = `
Object {
  "confidence": 0.99,
  "encoding": "UTF-8",
  "name": "UTF-8",
}
`;

exports[`iconv encode 4`] = `
Object {
  "data": Array [
    230,
    172,
    161,
    229,
    184,
    184,
    231,
    148,
    168,
    229,
    156,
    139,
    229,
    173,
    151,
    230,
    168,
    153,
    230,
    186,
    150,
    229,
    173,
    151,
    233,
    171,
    148,
    232,
    161,
    168,
  ],
  "type": "Buffer",
}
`;

exports[`iconv encode big5 1`] = `
Object {
  "confidence": 0.99,
  "encoding": "Big5",
  "name": "Big5",
}
`;

exports[`iconv encode big5 2`] = `
Object {
  "data": Array [
    166,
    184,
    177,
    96,
    165,
    206,
    176,
    234,
    166,
    114,
    188,
    208,
    183,
    199,
    166,
    114,
    197,
    233,
    170,
    237,
  ],
  "type": "Buffer",
}
`;

exports[`iconv encode big5 3`] = `
Object {
  "confidence": 0.99,
  "encoding": "Big5",
  "id": "big5",
  "name": "Big5",
}
`;

exports[`iconv encode big5 4`] = `
Object {
  "data": Array [
    166,
    184,
    177,
    96,
    165,
    206,
    176,
    234,
    166,
    114,
    188,
    208,
    183,
    199,
    166,
    114,
    197,
    233,
    170,
    237,
  ],
  "type": "Buffer",
}
`;
